src/main/java/mate/academy/springboot/datajpa/repository/specification/product/ProductCategoryInSpecificationProvider.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.springboot.datajpa.repository.specification.product;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductCategoryInSpecificationProvider implements SpecificationProvider<Product> {
+    private static final String FILTER_KEY = "categoryIn";
+    private static final String FIELD_NAME = "name";
+
+    @Override
+    public Specification<Product> getSpecification(String[] categories) {
+        return ((root, query, criteriaBuilder) -> {
+            CriteriaBuilder.In<String> predicate = criteriaBuilder.in(
+                    root.join("category", JoinType.LEFT)
+                            .get(FIELD_NAME));

```

Comment:
Do you want to get products without a category?

Reply:
Bro, it works well. I got filter by category when typing "categoryIn=%some_category".